schemainspect/pg/obj.py

```diff
@@ -1024,6 +1028,11 @@ def create_statement(self):
     def key(self):
         return "{} {}".format(self.object_type, self.identifier)
 
+    # Detects if the bug mentioned above is present
+    def is_domain_constraint_identifier_bug_present(self, identifier):
+        return " domain " not in identifier.lower()
```

Comment:
It is possible to construct a domain where the string " domain " is present in the identifier. As a rather contrived example: `create domain "the domain of all natural numbers" numeric check(VALUE >= 0 and floor(value)=value);` which I can add a comment on with `comment on constraint "the domain of all natural numbers_check" on domain public."the domain of all natural numbers" is 'totally contrived example';`. When running our sql I get this:
```-[ RECORD 1 ]----------------------------------------------------------------------------------------
object_type | constraint
identifier  | "the domain of all natural numbers_check" on public."the domain of all natural numbers"
comment     | totally contrived example```

Solving this for all possible inputs is probably impractical (the implementation would tend towards a sql parser at the limits). I think extending the check to be " on domain ", would be enough to turn it from a problem with edge cases, to a problem with corner cases. I think it's less likely for the extended phrase to show up in the identifiers; though I'm sure I could come up with an example.

Reply:
Good point. I have extended the check in my new commit.